### PR TITLE
Functional Tests : API - Basic Test

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Functional Tests : API - Basic Test
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Run the command `npm run test:functional:API`
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

Blocked by #31106 & #31107

### CODE REVIEW

File: tests/UI/pages/API/APIbasePage.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import {SuperAgentRequest} from 'superagent';
+
+use(chaiHttp);
+
+/**
+ * API parent page, contains functions that can be used on all API endpoints
+ * @class
+ * @extends CommonPage
+ */
+class APIBasePage extends CommonPage {
```
Comment (boubkerbribri) [inline tests/UI/pages/API/APIbasePage.ts]: As it's not a page, we can avoid call it a page and not extend CommonPage.
we can also add a new folder `helpers/api` and move this file.
And I'm not sure a class is the right approach to this kind tests.

File: tests/UI/campaigns/functional/API/01_basicTest.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Import utils
+import testContext from '@utils/testContext';
+
+import {expect} from 'chai';
+import {request, APIRequestContext} from '@playwright/test';
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/API/01_basicTest.ts]: You don't need to add '@playwright/test' the same types exists with playwright.

Comment (Progi1984) [reply tests/UI/campaigns/functional/API/01_basicTest.ts]: Fixed

File: tests/UI/package.json [hunk 3]
```diff
@@ -95,6 +97,7 @@
   "homepage": "https://github.com/PrestaShop/PrestaShop/tree/develop/tests/UI#readme",
   "dependencies": {
     "@faker-js/faker": "^7.6.0",
+    "@playwright/test": "^1.31.2",
```
Comment (boubkerbribri) [inline tests/UI/package.json]:
```suggestion
```

**Review by boubkerbribri — COMMENTED** [review]
Little comment

**Review by boubkerbribri — DISMISSED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Comment by nesrineabdmouleh** [comment]
Thanks @Progi1984

**Pull request merged**